### PR TITLE
Improve language and appearance of 'no JavaScript' warning

### PR DESCRIPTION
**Pull request by n0toose**
- Add two footers, so that the message is not as distracting without
  affecting instances using own branding (i.e. Blender, Codeberg, etc.)
- Explain why things work better with JavaScript, so as to allow the
  user to make a calculated decision.
- "hints" stands in for "tooltips", as the term should be
  more user- and translator-friendly
- Add new site-specific warning to Installer page, so as to further
  prevent new users that are trying out the software locally from
  getting confused. The old "This website requires JavaScript" was
  used.

### CODE REVIEW

**Comment by n0toose** [comment]
Here's a screenshot:

image

File: templates/base/footer.tmpl [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 	{{template "custom/body_outer_post" .}}
 
+	<noscript>{{template "base/footer_noscript" .}}</noscript>
```
Comment (delvh) [inline templates/base/footer.tmpl]: I think it would be best to have the `<noscript>` inside the new template instead of here.

Comment (n0toose) [reply templates/base/footer.tmpl]: Are you sure? I would assume that putting the entire footer inside of a `<noscript>` would render a footer without any contents on the end user's view.

I'm far from a web developer expert, but is this definitely a clean solution to the problem?

Comment (n0toose) [reply templates/base/footer.tmpl]: Yeah, as my gut feeling said, this renders an additional separator for everyone that does use JavaScript.

image

**Comment by jolheiser** [comment]
I like the HTML update and translation, but wouldn't it be better to leave it up at the top so it's more immediately noticeable?

**Comment by n0toose** [comment]
> I like the HTML update and translation, but wouldn't it be better to leave it up at the top so it's more immediately noticeable?

If you want to be 100% sure that the user will see that on their first visit, the answer is yes. That's how basically every site ever does it, and it's good to have a warning in case someone tries to use the site for the event that stuff doesn't work.

The reason why I chose to go up against such a de facto principle was because the top bar, I think, is "functional enough" without JavaScript as it is. Most people that would use Gitea would be "repeat users" and would probably see the footer at least once in their lives, and testing the software without JavaScript made me think that this warning is kind of obnoxious - it's the kind of thing you'd see in an average modern site - but when I have to quickly glance at the top bar and click on something, it feels mildly distracting and it feels like I'm being kind of shamed for that with the current warning.

As things are right now, the buttons on the top navbar work and it's just the dropdown menus that may look a bit weird right now. A person who may use an element that requires a dropdown menu will probably have scrolled down to the bottom / have seen the homepage at least once.

**Comment by n0toose** [comment]
> I like the HTML update and translation, but wouldn't it be better to leave it up at the top so it's more immediately noticeable?

Now that you mention it, I am a bit worried about the *Installation* page, which shows dropdown menus that do not work without JavaScript, and it threw me off as I was testing this very change. I will take a more careful look.

File: templates/base/footer.tmpl (hunk 1 above)
Comment (delvh) [reply templates/base/footer.tmpl]: I meant not inside the footer, but inside the new template.
I don't see why this should change anything?

**Comment by n0toose** [comment]
I think that this edge case was dealt with and I do not see myself expanding the scope of my PR any further.

Should I add an empty file under `templates/custom/footer_noscript.tmpl` to allow people to customize that part?

### Screenshot

https://github.com/go-gitea/gitea/assets/30193966/b9d576ee-f10a-45e0-a1d4-47761665f472)

**Comment by lunny** [comment]
I couldn't find the changes necessary. Since Gitea cannot guarantee consistency without Javascript, a block error like before is the best choice to not confuse users.

If we want to support non-javascript, we need to have a whole redesign.

**Review by lunny — CHANGES_REQUESTED** [review]
https://github.com/go-gitea/gitea/pull/28349#issuecomment-1840123335

**Comment by n0toose** [comment]
> I couldn't find the changes necessary. Since Gitea cannot guarantee consistency without Javascript,

The error is IMO too generic for a user of NoScript, "This website requires JavaScript" is something that I associate with popular websites that show ads and contain other fancy elements, when you can just read its contents without it. 

> If we want to support non-javascript, we need to have a whole redesign.

Supporting no-JavaScript setups is not the intention, being transparent as to why JavaScript is necessary was. "Well yes, you can theoretically use it without JavaScript, but your experience will be partially trash and that's why you should enable it" is what I wish to communicate.

Perhaps I went too far with the additional UI changes. Should I remove everything else and just keep the text change, maybe replacing the "works best with JavaScript" with "requires JavaScript" and *then* following up with the explanation ("... why you should enable it")?

**Comment by lunny** [comment]
> Claiming to support no-JavaScript setups is not the intention, being transparent as to why JavaScript is necessary was. "Well yes, you can theoretically use it without JavaScript, but your experience will be partially trash and that's why you should enable it" is what I wish to communicate.

But work correctly without javascript is not in the design targets. So we cannot say that. If we want to have that permission, many works need to be done.

**Comment by n0toose** [comment]
> But work correctly without javascript is not in the design targets. So we cannot say that. If we want to have that permission, many works need to be done.

Got it, thanks for the explanation. I will modify my existing changes later.

**Comment by wxiaoguang** [comment]
I also prefer to keep things simple.

In modern days, the people who explicitly disable JavaScript should already have the knowledge that "many websites do not work". Just like a man who keeps using an old feature phone, when they sees a QR code, they should already know that "Oh, my phone doesn't work with it". The QR code itself doesn't need to say "please use a smart phone".

File: templates/install.tmpl [hunk 2]
```diff
@@ -8,6 +8,10 @@
 			<div class="ui attached segment">
 				{{template "base/alert" .}}
 
+				{{/* Drop-down menus do not currently work well without JavaScript, only there to prevent "bad first impressions". */}}
+				{{/* TODO: Remove this together with "enable_javascript" string when that is fixed. */}}
```
Comment (wxiaoguang) [inline templates/install.tmpl:12]: The "install" page is impossible to work without JS. There are a lot of JS code for the "install" page. See "install.js"

Comment (n0toose) [reply templates/install.tmpl:12]: Yeah, I found out myself, which is why I tried to add an extra, super loud warning specifically for the install page.

Thanks for the review, I'll try again later.

**Comment by wxiaoguang** [comment]
I will close this PR since it is stale for long time.

Actually, Gitea heavily depends on JS in modern days, and we should tell users that "JS is required" instead of trying to make Gitea work without JS.

Feel free to reopen if there would be improvements.

**Comment by n0toose** [comment]
Don't worry, I fully understand and agree. I find it unfortunate that our goals somewhat diverged too much, but I appreciate your courteous reply.

Have a great weekend.

**Pull request closed without merging**